cmd/thv-proxyrunner/app/run.go

```diff
@@ -0,0 +1,411 @@
+   with './' or '../' to build and run local Go projects.
+
+The container will be started with the specified transport mode and
+permission profile. Additional configuration can be provided via flags.`,
+	Args: cobra.MinimumNArgs(1),
+	RunE: runCmdFunc,
+	// Ignore unknown flags to allow passing flags to the MCP server
+	FParseErrWhitelist: cobra.FParseErrWhitelist{
+		UnknownFlags: true,
+	},
+}
+
+var (
+	runTransport         string
+	runName              string
+	runHost              string
+	runPort              int
+	runTargetPort        int
+	runTargetHost        string
+	runPermissionProfile string
+	runEnv               []string
+	runForeground        bool
+	runVolumes           []string
+	runSecrets           []string
+	runAuthzConfig       string
+	runAuditConfig       string
+	runEnableAudit       bool
+	runK8sPodPatch       string
+	runCACertPath        string
+	runVerifyImage       string
+
+	// OpenTelemetry flags
+	runOtelEndpoint                    string
+	runOtelServiceName                 string
+	runOtelSamplingRate                float64
+	runOtelHeaders                     []string
+	runOtelInsecure                    bool
+	runOtelEnablePrometheusMetricsPath bool
+	runOtelEnvironmentVariables        []string
+
+	// Network isolation flag
+	runIsolateNetwork bool
+)
+
+func init() {
+	runCmd.Flags().StringVar(&runTransport, "transport", "", "Transport mode (sse, streamable-http or stdio)")
+	runCmd.Flags().StringVar(&runName, "name", "", "Name of the MCP server (auto-generated from image if not provided)")
+	runCmd.Flags().StringVar(&runHost, "host", transport.LocalhostIPv4, "Host for the HTTP proxy to listen on (IP or hostname)")
+	runCmd.Flags().IntVar(&runPort, "port", 0, "Port for the HTTP proxy to listen on (host port)")
+	runCmd.Flags().IntVar(&runTargetPort, "target-port", 0,
+		"Port for the container to expose (only applicable to SSE or Streamable HTTP transport)")
+	runCmd.Flags().StringVar(
+		&runTargetHost,
+		"target-host",
+		transport.LocalhostIPv4,
+		"Host to forward traffic to (only applicable to SSE or Streamable HTTP transport)")
+	runCmd.Flags().StringVar(
+		&runPermissionProfile,
+		"permission-profile",
+		permissions.ProfileNetwork,
+		"Permission profile to use (none, network, or path to JSON file)",
+	)
+	runCmd.Flags().StringArrayVarP(
+		&runEnv,
+		"env",
+		"e",
+		[]string{},
+		"Environment variables to pass to the MCP server (format: KEY=VALUE)",
+	)
+	runCmd.Flags().BoolVarP(&runForeground, "foreground", "f", false, "Run in foreground mode (block until container exits)")
+	runCmd.Flags().StringArrayVarP(
+		&runVolumes,
+		"volume",
+		"v",
+		[]string{},
+		"Mount a volume into the container (format: host-path:container-path[:ro])",
+	)
+	runCmd.Flags().StringArrayVar(
+		&runSecrets,
+		"secret",
+		[]string{},
+		"Specify a secret to be fetched from the secrets manager and set as an environment variable (format: NAME,target=TARGET)",
+	)
+	runCmd.Flags().StringVar(
+		&runAuthzConfig,
```

Comment:
On one hand, I suspect we need some way to set auth config for k8s, but is it going to be a file path?